irohad/ametsuchi/impl/postgres_query_executor.hpp

```diff
@@ -146,20 +146,60 @@ namespace iroha {
        * Execute query which returns list of transactions
        * uses pagination
        * @param query - query object
+       * @param qry_checker - fallback checker of the query, needed in some
+       * unclear cases
        * @param related_txs - SQL query which returns transaction relevant
        * to this query
        * @param applier - function which accepts SQL
        * and returns another function which executes that query
        * @param perms - permissions, necessary to execute the query
        * @return Result of a query execution
        */
-      template <typename Query, typename QueryApplier, typename... Permissions>
+      template <typename Query,
+                typename QueryChecker,
+                typename QueryApplier,
+                typename... Permissions>
       QueryExecutorResult executeTransactionsQuery(
           const Query &query,
+          QueryChecker &&qry_checker,
           const std::string &related_txs,
           QueryApplier applier,
           Permissions... perms);
 
+      /**
+       * Check if entry with such key exists in the database
+       * @tparam ReturnValueType - type of the value to be returned in the
+       * underlying query
+       * @param table_name - name of the table to be checked
+       * @param key_name - name of the table attritute, against which the search
+       * is performed
+       * @param selected_value_name - name of the value, which is to be returned
+       * from the search (attribute with such name is to exist)
+       * @param actual value of the key attribute
+       * @return true, if entry with such value of the key attribute exists,
+       * false otherwise
+       */
+      template <typename ReturnValueType>
+      bool existsInDb(const std::string &table_name,
+                      const std::string &key_name,
+                      const std::string &value_name,
+                      const std::string &value) const;
+
+      struct QueryFallbackCheckResult {
+        QueryFallbackCheckResult() = default;
+        QueryFallbackCheckResult(size_t error_code, std::string &&error_message)
+            : contains_error_{true},
+              error_code_{error_code},
+              error_message_{std::move(error_message)} {}
+
+        explicit operator bool() const {
+          return contains_error_;
+        }
+        bool contains_error_ = false;
+        size_t error_code_ = 0;
+        std::string error_message_ = "";
```

Comment:
```suggestion
        shared_model::interface::ErrorQueryResponse::ErrorMessageType error_message = "";
```
since these fields are public